### ISSUE TITLE
Does allennlp support "sample_softmax" with both dense scheduler and sparse scheduler? 

### ISSUE THREAD

**Issue by robbine** [allenai/allennlp#3081]
Here is a transformer-xl implementation.
https://github.com/kimiyoung/transformer-xl/blob/master/pytorch/train.py
Is it possible to incorporate those optimizers as well as schedulers into allennlp ?
I think sparse scheduler is really helpful when the token embedding matrix is extremely large in which case we can use
`scheduler_sparse = optim.lr_scheduler.CosineAnnealingLR` to alleviate training cost.
Thanks!

**Comment by joelgrus** [comment]
if they're just standard pytorch optimizers / schedulers, it should be simple enough to add them as "registered" options, PR welcome

**Comment by DeNeutoy** [comment]
These can be registered independently, and don't need to live in the library.
